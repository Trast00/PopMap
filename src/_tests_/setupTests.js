import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListCountry = [
  {
    name: 'test country 1',
    population: 1000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test1',
  },
  {
    name: 'test country 2',
    population: 2000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test2',
  },
];

export const fetchDataMock = () => {
  fetchMock.getOnce('https://restcountries.com/v2/all', {
    headers: { 'content-type': 'application/json' },
    body: testListCountry,
  });
};
