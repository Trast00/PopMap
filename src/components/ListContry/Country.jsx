import React from 'react'

const Country = (props) => {
  const {country} = props
  const {name, population, flags} = country
  return (
    <div className='flex-center column country'>
      <img src={flags.svg} alt={name + `Country Images`} />
      <div className="column detail">
        <h4>{name}</h4>
        <p>{population}</p>
      </div>
    </div>
  )
}

export default Country
