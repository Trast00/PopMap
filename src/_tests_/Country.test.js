import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { testListCountry } from './setupTests.js';
import Country from '../components/ListContry/Country.jsx';

describe('test render', () => {
  test('List Rocket should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
          <Country country={testListCountry[0]} />
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})