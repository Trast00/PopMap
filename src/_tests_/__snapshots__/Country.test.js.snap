// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<div
  className="flex-center column country"
>
  <img
    alt="test country 1Country Images"
    src="test image url"
  />
  <div
    className="column detail-wrapper"
  >
    <h4>
      test country 1
    </h4>
    <p>
       
      1000
    </p>
  </div>
</div>
`;
